lxd/storage/drivers/driver_btrfs_volumes.go

```diff
@@ -1408,9 +1410,35 @@ func (d *btrfs) UnmountVolumeSnapshot(snapVol Volume, op *operations.Operation)
 	return forceUnmount(snapPath)
 }
 
-// VolumeSnapshots returns a list of snapshots for the volume (in no particular order).
+// VolumeSnapshots returns a list of snapshots for the volume (ordered by subvolume ID).
+// Since the subvolume ID is incremental, this also represents the order of creation.
 func (d *btrfs) VolumeSnapshots(vol Volume, op *operations.Operation) ([]string, error) {
-	return genericVFSVolumeSnapshots(d, vol, op)
+	args := []string{"subvolume", "list", GetPoolMountPath(vol.pool)}
+
+	output, err := shared.RunCommand("btrfs", args...)
+	if err != nil {
+		return nil, err
+	}
+
+	var snapshotNames []string
+
+	scanner := bufio.NewScanner(strings.NewReader(output))
+
+	for scanner.Scan() {
+		fields := strings.Fields(scanner.Text())
+
+		if len(fields) != 9 {
+			continue
+		}
+
+		if !strings.HasPrefix(fields[8], fmt.Sprintf("%s-snapshots/%s", vol.volType, vol.name)) {
```

Comment:
Minor nit, but the `fmt.Sprintf` could be moved out of the for loop since it should be identical for every iteration. Dunno if the compiler is smart enough to do that itself though.